subsys/net/ip/Kconfig

```diff
@@ -109,6 +109,15 @@ config NET_DEBUG_TCP
 	help
 	Enables TCP handler output debug messages
 
+config NET_TCP_BACKLOG_SIZE
+	int "Number of simultaneous incoming TCP connections"
+	depends on NET_TCP
+	default 2
+	range 2 128
```

Comment:
I think we could start the range from 1, just in case the end user wants to optimize memory consumption.